gnovm/pkg/doctest/exec.go

```diff
@@ -0,0 +1,246 @@
+package doctest
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync"
+
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	bft "github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/db/memdb"
+	"github.com/gnolang/gno/tm2/pkg/log"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	authm "github.com/gnolang/gno/tm2/pkg/sdk/auth"
+	bankm "github.com/gnolang/gno/tm2/pkg/sdk/bank"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/gnolang/gno/tm2/pkg/store"
+	"github.com/gnolang/gno/tm2/pkg/store/dbadapter"
+	"github.com/gnolang/gno/tm2/pkg/store/iavl"
+)
+
+// Option constants
+const (
+	IGNORE       = "ignore"       // Do not run the code block
+	SHOULD_PANIC = "should_panic" // Expect a panic
+	ASSERT       = "assert"       // Assert the result and expected output are equal
+)
+
+// GetStdlibsDir returns the path to the standard libraries directory.
+func GetStdlibsDir() string {
+	_, filename, _, ok := runtime.Caller(0)
```

Comment:
I made it search for the location in the gnovm directory like that because other methods couldn't recognize the stdlib.

When using other methods, I couldn't use such packages. There might be better ways to handle this, but I can't think of any for now.